src/NuGet.Clients/NuGet.Tools/DataStreamFromComStream.cs

```diff
@@ -38,6 +39,7 @@ public override bool CanWrite
 
         public override void Flush()
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
What is the current behavior if not an UI thread? What are the current scenarios that could start failing with this change?

Reply:
These are changes that the analyzers recommended. 
There are no scenarios where this is not on the UI thread. 
All COM calls happen on the UI thread and this class specifically deals only with COM objects. 

If we were running on any other thread than the UI thread we would've deadlocked already.